src/block_reporting.js

```diff
@@ -0,0 +1,21 @@
+import * as Blockly from 'blockly/core';
+import {Colours} from '../core/colours.js';
+
+export function reportValue(id, value) {
+  const block = Blockly.getMainWorkspace().getBlockById(id) ||
+      Blockly.getMainWorkspace().getFlyout().getWorkspace().getBlockById(id);
```

Comment:
Did you test dragging a block from the flyout and then clicking the block in the flyout? I think in the latest version the first block dragged from the flyout has the same ID as the block in the flyout, because of some unit tests code.org has that depended on that functionality. So I'm thinking clicking the block in the flyout may actually show the reporter on the block in the workspace.

If that's the case... I think the best solution is probably hacking at the continuous toolbox more. Unless we can pass in the ID of the workspace in addition to the ID of the block.

Reply:
Oh that's awful, yes that is what happens. And by hack at the continuous toolbox, you mean have it override/change the ID on drag?

Reply:
Yes! But I think looking at your other PR, we should just be able to pass in the workspace ID as well?

Reply:
I don't think so, `this.workspace` is just initialized to the result of `inject` in scratch-gui, it doesn't contextually change between the flyout/main workspace.

Reply:
Darn yeah. Ok then, I think it's easier for us to make the change on the Scratch side than to make code.org fix their tests (mostly because we reverted the change because it broke their tests).

Reply:
Yeah that seems fine, I'll take a look at doing that. Changing the block ID at creation shouldn't have scary effects elsewhere right?

Reply:
It might break some of scratch's tests :P But yeah should be fine. We did it in core already and the only issue it caused was breaking the tests in code.org

Reply:
Overrode the method introduced in https://github.com/google/blockly/pull/8065 to address this. Anchoring of the reporter bubble will be broken until we update to v11, but the extra method won't hurt anything in the meantime.